### ISSUE TITLE
state does not update after running terraform apply

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#1424]
_This issue was originally opened by @YiyangXu as hashicorp/terraform#15807. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi there,
I am in a situation when terraform apply showed a correct behavior in terminal but the state does not actually update and the change did not deployed in aws. 

```sh
$ terraform apply
```
```
aws_dynamodb_table.session-table: Refreshing state... (ID: account-service-session-db)
aws_dynamodb_table.session-table: Modifying... (ID: account-service-session-db)
  attribute.#:               "2" => "4"
  attribute.1670162472.name: "" => "lastUpdateDate"
  attribute.1670162472.type: "" => "N"
  attribute.2901025699.name: "accountID" => "accountID"
  attribute.2901025699.type: "N" => "N"
  attribute.646540669.name:  "token" => "token"
  attribute.646540669.type:  "S" => "S"
  attribute.987920283.name:  "" => "createDate"
  attribute.987920283.type:  "" => "N"
aws_dynamodb_table.session-table: Modifications complete (ID: account-service-session-db)

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

```sh
$ terraform show
```
```
aws_dynamodb_table.session-table:
  id = account-service-session-db
  arn = arn:aws:dynamodb:us-west-2:113514851486:table/account-service-session-db
  attribute.# = 2
  attribute.2901025699.name = accountID
  attribute.2901025699.type = N
  attribute.646540669.name = token
  attribute.646540669.type = S
  global_secondary_index.# = 0
  hash_key = token
  local_secondary_index.# = 0
  name = account-service-session-db
  range_key = accountID
  read_capacity = 5
  write_capacity = 5
```

**Comment by bflad** [comment]
This has been released in [version 1.12.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1120-march-23-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by alambike** [comment]
Seeing same behaviour with **aws_lightsail_instance** resource. After recreating instance the outputs show previous instance info:
After executing `terraform refresh` the info about current instance is updated correctly.

```
terraform version: 0.11.7
aws provider version: 1.26.0
```
